org-node/src/client.rs

```diff
@@ -277,70 +523,24 @@ impl Client {
-                        }
-                        Ok(tracked) => {
-                            let response = if tracked {
-                                tracing::debug!(target: "org-node", "Tracking relationship for project {} and peer {} created", urn, peer.peer_id);
-
-                                Some(peer.peer_id)
-                            } else {
-                                tracing::debug!(target: "org-node", "Tracking relationship for project {} already exists", urn);
-
-                                None
-                            };
-                            let cfg = api.protocol_config().replication;
-                            let peer_id = peer.peer_id;
-
-                            // Try to fetch the project. Even though the project might have already
-                            // been tracked, it has not been replicated.
-                            let result = api
-                                .using_storage({
-                                    let urn = urn.clone();
-
-                                    move |s| {
-                                        Client::fetch_project(
-                                            &urn,
-                                            peer_id,
-                                            peer.seen_addrs.to_vec(),
-                                            s,
-                                            cfg,
-                                        )
-                                    }
-                                })
-                                .await?;
-
-                            // If we fail to replicate, try the next peer.
-                            if let Err(err) = result {
-                                tracing::error!(target: "org-node", "Failed to replicate {} from {}: {}", urn, peer_id, err);
-                                continue;
-                            }
-
-                            let Head { remote, branch } = Client::get_project_head(&urn, api)
-                                .await?
-                                .expect("a project that was just replicated should exist");
-
-                            // Fetching doesn't automatically set the repository head, we have to do it
-                            // manually. We set the head to the default branch of the project
-                            // maintainer.
-                            match Client::set_head(&urn, &remote, &branch, paths) {
-                                Err(Error::SetHead(err)) => {
-                                    tracing::error!(target: "org-node", "Error setting head: {}", err);
-                                }
-                                Err(err) => return Err(err),
-                                Ok(()) => {}
-                            }
-                            Client::sign_refs(urn, api).await?;
+                if let Some(peer) = peers.next().await {
```

Comment:
Hmm I don't think that's right: the while loop is there because we are attempting peers one by one until the first one succeeds to replicate. With an `if`, we would only be trying the first peer we find.

Reply:
3b77898 reverts to using while loop; I think what I was missing originally was continuing on an error for the match arm, such that the loop does not return an early early, breaking the loop.